java/mapreduce/src/java/org/apache/orc/mapred/OrcInputFormat.java

```diff
@@ -151,4 +153,26 @@ public static void setSearchArgument(Configuration conf,
     return new OrcMapredRecordReader<>(file, buildOptions(conf,
         file, split.getStart(), split.getLength()));
   }
+
+  /**
+   * Filter out the 0 byte files, so that we don't generate splits for the
+   * empty ORC files.
+   * @param job the job configuration
+   * @return a list of files that need to be read
+   * @throws IOException
+   */
+  protected FileStatus[] listStatus(JobConf job) throws IOException {
+    FileStatus[] result = super.listStatus(job);
+    List<FileStatus> ok = new ArrayList<>(result.length);
```

Comment:
Instead of checking this after retrieving all FileStatus objects, it will be better if a PathFilter can be passed to listStatus() so that we will only get non-zero files. Getting 1000s of 0 length files and filtering here seems wasteful.

Reply:
That would involve doing a second getStatus. Filters are only given the name.

Reply:
Make sense. Just noticed filter gets applied after listStatus anyway.